hudi-common/src/main/java/org/apache/hudi/metadata/HoodieBackedTableMetadata.java

```diff
@@ -292,180 +325,58 @@ private static ArrayList<ArrayList<String>> partitionKeysByFileSlices(List<Strin
-
-    metrics.ifPresent(m -> m.updateMetrics(HoodieMetadataMetrics.BASEFILE_READ_STR, readTimer.endTimer()));
-
-    // Iterate over all provided log-records, merging them into existing records
-    logRecords.values().forEach(logRecord ->
-        records.merge(
-            logRecord.getRecordKey(),
-            logRecord,
-            (oldRecord, newRecord) -> {
-              HoodieMetadataPayload mergedPayload = newRecord.getData().preCombine(oldRecord.getData());
-              return mergedPayload.isDeleted() ? null : new HoodieAvroRecord<>(oldRecord.getKey(), mergedPayload);
-            }
-        ));
-
-    timings.add(timer.endTimer());
-    return records;
-  }
-
-  @SuppressWarnings("unchecked")
-  private Map<String, HoodieRecord<HoodieMetadataPayload>> fetchBaseFileRecordsByKeys(HoodieSeekingFileReader reader,
-                                                                                      List<String> sortedKeys,
-                                                                                      boolean fullKeys,
-                                                                                      String partitionName) throws IOException {
-    Map<String, HoodieRecord<HoodieMetadataPayload>> result;
-    try (ClosableIterator<HoodieRecord<?>> records = fullKeys
-        ? reader.getRecordsByKeysIterator(sortedKeys)
-        : reader.getRecordsByKeyPrefixIterator(sortedKeys)) {
-      result = toStream(records)
-          .map(record -> {
-            GenericRecord data = (GenericRecord) record.getData();
-            // populateMetaFields is hardcoded to false for the metadata table so key must be extracted from the `key` field
-            String recordKey = (String) data.get(HoodieMetadataPayload.KEY_FIELD_NAME);
-            return Pair.of(recordKey, composeRecord(data, recordKey, partitionName));
-          })
-          .collect(Collectors.toMap(Pair::getKey, Pair::getValue));
-    }
-    return result;
-  }
-
-  private HoodieRecord<HoodieMetadataPayload> composeRecord(GenericRecord avroRecord, String recordKey, String partitionName) {
-    return new HoodieAvroRecord<>(new HoodieKey(recordKey, partitionName),
-        new HoodieMetadataPayload(avroRecord, 0L), null);
-  }
-
-  /**
-   * Create a file reader and the record scanner for a given partition and file slice
-   * if readers are not already available.
-   *
-   * @param partitionName - Partition name
-   * @param slice         - The file slice to open readers for
-   * @return File reader and the record scanner pair for the requested file slice
-   */
-  private Pair<HoodieSeekingFileReader<?>, HoodieMetadataLogRecordReader> getOrCreateReaders(String partitionName, FileSlice slice) {
-    if (reuse) {
-      Pair<String, String> key = Pair.of(partitionName, slice.getFileId());
-      return partitionReaders.get().computeIfAbsent(key, ignored -> openReaders(partitionName, slice));
-    } else {
-      return openReaders(partitionName, slice);
-    }
```

Comment:
Is the `reuse` behavior handled in a different way though the FGReader-based MDT reading?